spec/jobs/regular/auto_manage_channel_memberships_spec.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe Jobs::AutoManageChannelMemberships do
+  let(:user) { Fabricate(:user, last_seen_at: 15.minutes.ago) }
+  let(:category) { Fabricate(:category, user: user) }
+  let(:channel) { Fabricate(:chat_channel, auto_join_users: true, chatable: category) }
+
+  describe 'queues batches to automatically add users to a channel' do
+    it 'queues a batch for users with channel access' do
+      assert_batches_enqueued(channel, 1)
+    end
+
+    it "does nothing when the channel doesn't exist" do
+      assert_batches_enqueued(ChatChannel.new(id: -1), 0)
+    end
+
+    it 'does nothing when the chatable is not a category' do
+      same_id = 99
+      another_category = Fabricate(:category, id: same_id)
+      dm_channel = Fabricate(:direct_message_channel, id: same_id)
+      channel.update!(chatable: dm_channel)
+
+      assert_batches_enqueued(channel, 0)
+    end
```

Comment:
@romanrizzi do you remember why those two have to have the same id in this test? Specifying an id when creating a record makes this test flaky - you're never guaranteed that given id is available.

Reply:
It's to make sure the `chatable_type` is used when finding the channel we'll auto-join users to. We can remove it if it's causing problems.